### PR TITLE
Check startup script when check if configmap has beed updated

### PR DESCRIPTION
**Pull request by KanShiori**
### What problem does this PR solve?

Check startup script when check if configmap has beed updated

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [x] Manual test 
Upgrade operator v1.1.7 to v1.2.4:
* If set `clusterDomain` in TidbCluster, `PD`,`TiKV` and `TiDB` roll update. Because of PR #4123
* If don't set `clusterDomain` in TidbCluster, `PD`,`TiKV` and `TiDB` don't roll update.
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Roll update `PD`, `TiKV`, `TiDB` if set `clusterDomain`.
```

### CODE REVIEW

File: pkg/manager/member/config.go [hunk 1]
```diff
@@ -52,6 +53,18 @@ func updateConfigMap(old, new *corev1.ConfigMap) (bool, error) {
 		}
 	}
```
Comment (july2993) [inline pkg/manager/member/config.go:55]: how about just
```
if !reflect.DeepEqual(old.Data, new.Data) {
    return false nil
}

return true, nil

```

**Review by july2993 — APPROVED** [review]

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Pull request merged**